modules/AlphaWalletFoundation/AlphaWalletFoundation/Tokens/Helpers/CallSmartContractFunction.swift

```diff
@@ -88,13 +88,13 @@ fileprivate var smartContractCallsCache = AtomicDictionary<String, (promise: Pro
 private let callSmartContractQueue = DispatchQueue(label: "com.callSmartContractQueue.updateQueue")
 //`shouldDelayIfCached` is a hack for TokenScript views
 //TODO should trap 429 from RPC node
-public func callSmartContract(withServer server: RPCServer, contract: AlphaWallet.Address, functionName: String, abiString: String, parameters: [AnyObject] = [], shouldDelayIfCached: Bool = false, queue: DispatchQueue? = nil) -> Promise<[String: Any]> {
+public func callSmartContract(withServer server: RPCServer, contract: AlphaWallet.Address, functionName: String, abiString: String, parameters: [AnyObject] = [], shouldDelayIfCached: Bool = false, queue: DispatchQueue? = nil, useCashed: Bool = true) -> Promise<[String: Any]> {
     let timeout: TimeInterval = 60
     //We must include the ABI string in the key because the order of elements in a dictionary when serialized in the string is not ordered. Parameters (which is ordered) should ensure it's the same function
     let cacheKey = "\(contract).\(functionName) \(parameters) \(server.chainID)"
     let ttlForCache: TimeInterval = 10
     let now = Date()
-    if let (cachedPromise, cacheTimestamp) = smartContractCallsCache[cacheKey] {
+    if let (cachedPromise, cacheTimestamp) = smartContractCallsCache[cacheKey], useCashed {
```

Comment:
Faster to flip the conditions order so `useCashed` comes before `let`? Faster. But is it less readable?

Reply:
will do